Update project to .NET 8.0 and migrate to ReqnRoll

HowardvanRooijen opened the pull request:
- Target framework updated to .NET 8.0 in project files.
- Upgraded `Endjin.RecommendedPractices.GitHub` to 2.1.18.
- Updated `Microsoft.Extensions.DependencyInjection` to 8.0.*.
- Replaced `Corvus.Testing.SpecFlow.NUnit` with `Corvus.Testing.ReqnRoll.NUnit` (v4.0.5).
- Removed `specflow.json` and added `reqnroll.json` for new testing framework configuration.
- Updated `Newtonsoft.Json` version from 13.0.1 to 13.0.3.
- Refactored namespaces and improved null safety in relevant files.
- Updated comments for clarity and consistency across files.
- Modified solution settings to include new resource dictionary for grammar and spelling.

HowardvanRooijen committed:
Update project to .NET 8.0 and migrate to ReqnRoll

- Target framework updated to .NET 8.0 in project files.
- Upgraded `Endjin.RecommendedPractices.GitHub` to 2.1.18.
- Updated `Microsoft.Extensions.DependencyInjection` to 8.0.*.
- Replaced `Corvus.Testing.SpecFlow.NUnit` with `Corvus.Testing.ReqnRoll.NUnit` (v4.0.5).
- Removed `specflow.json` and added `reqnroll.json` for new testing framework configuration.
- Updated `Newtonsoft.Json` version from 13.0.1 to 13.0.3.
- Refactored namespaces and improved null safety in relevant files.
- Updated comments for clarity and consistency across files.
- Modified solution settings to include new resource dictionary for grammar and spelling.

JamesDawson committed:
Migrate to ZeroFailed

jongeorge1 commented on Solutions/Corvus.Extensions.Newtonsoft.Json.Specs/JsonExtensionsSpecsSteps.cs:
```diff
@@ -37,12 +37,14 @@ public JsonExtensionsSpecsSteps(FeatureContext featureContext, ScenarioContext s
             this.scenarioContext = scenarioContext;
-            this.propertyBagFactory = ContainerBindings.GetServiceProvider(featureContext).GetService<IPropertyBagFactory>();
-            this.jnetPropertyBagFactory = ContainerBindings.GetServiceProvider(featureContext).GetService<IJsonNetPropertyBagFactory>();
+            this.propertyBagFactory = ContainerBindings.GetServiceProvider(featureContext).GetService<IPropertyBagFactory>()!;
```
Better to use `GetRequiredService` here (and the other similar places in this file) instead of ignoring the possibility of a null result.

HowardvanRooijen replied:
Done

jongeorge1 commented on Solutions/Corvus.Json.sln.DotSettings:
What's this?

HowardvanRooijen replied:
Spelling dictionary. Have deleted

HowardvanRooijen committed:
Refactor service retrieval and improve readability

Updated service retrieval from `GetService<T>()` to `GetRequiredService<T>()` to ensure non-null services. Simplified the formatting of the `Bag` property getter for better readability.

HowardvanRooijen committed:
Merge branch 'features/update-dotnet-8' of https://github.com/corvus-dotnet/Corvus.Extensions.Newtonsoft.Json into features/update-dotnet-8

HowardvanRooijen committed:
Remove grammar checking exception entry from settings

HowardvanRooijen committed:
Refactor service retrieval and improve code readability

Updated service retrieval in `JsonExtensionsSpecsSteps` to use `GetService<T>()!` for better null handling. Adjusted method signatures and logic in `JsonSerializerSettingsProviderServiceCollectionExtensions` for improved readability and service registration checks. Minor syntax improvements for object initialization and nullable type handling were also made.
